### PR TITLE
feat(test): Add ECS CreateServerGroupTests for task definition input

### PR DESCRIPTION
**Pull request by paragbhingre**
This PR contains integration tests for Create Server Group Operation for the task definition type as input.

There are total 2 tests:
1. Create server group operation for the task definition type as input, launch type as FARGATE with legacy target group.
2. Create server group operation for the  task definition type as input, launch type as FARGATE with new target group.

Related to : [Clouddriver integration test](https://github.com/spinnaker/clouddriver/pull/4827)

### CODE REVIEW

File: clouddriver-ecs/src/integration/resources/testoperations/createServerGroupOperation-inputs-fargate-NewTargetGroup.json [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  "stack": "integTestStack",
+  "freeFormDetails" : "detailTest",
+  "targetGroupMappings": [
+    {
+      "containerName" : "main",
+      "containerPort" : 80,
+      "targetGroup" : "spin-fargate-instanceTG"
+    }
+  ],
+  "source" : {
```
Comment (allisaurus) [inline clouddriver-ecs/src/integration/resources/testoperations/createServerGroupOperation-inputs-fargate-NewTargetGroup.json:33]: what happens if we don't include this `source` arg, does it fail?

**Comment by allisaurus** [comment]
@paragbhingre since this has a `feat(test)` prefix, can we add "ECS" to the commit msg somewhere to indicate it's provider affiliation? E.G., "feat(test): Add ECS CreateServerGroup integ tests for task definition input"

File: clouddriver-ecs/src/integration/java/com/netflix/spinnaker/clouddriver/ecs/test/CreateServerGroupSpec.java [hunk 2]
```diff
@@ -66,6 +66,76 @@ public void createServerGroupOperationTest() throws IOException {
...
 
+  @DisplayName(
+      ".\n===\n"
+          + "Given description w/ task def inputs, FARGATE launch type, and legacy target group fields, "
+          + "successfully submit createServerGroup operation"
+          + "\n===")
+  @Test
+  public void createSGOWithInputsFargateLegacyTargetGroupTest() throws IOException {
+    /**
+     * TODO (pbhingre): Ideally this test would go further and actually assert that the resulting
```
[the remark this answers is not in the document]
Comment (paragbhingre) [reply clouddriver-ecs/src/integration/java/com/netflix/spinnaker/clouddriver/ecs/test/CreateServerGroupSpec.java]: Done.

File: clouddriver-ecs/src/integration/java/com/netflix/spinnaker/clouddriver/ecs/test/CreateServerGroupSpec.java [hunk 3]
```diff
@@ -66,6 +66,76 @@ public void createServerGroupOperationTest() throws IOException {
...
+     * ecs:create-service call is formed as expected, but for now, it asserts that the given
+     * operation is correctly validated and submitted as a task.
+     */
+
+    // given
+    String url = getTestUrl("/ecs/ops/createServerGroup");
```
[the remark this answers is not in the document]
Comment (paragbhingre) [reply clouddriver-ecs/src/integration/java/com/netflix/spinnaker/clouddriver/ecs/test/CreateServerGroupSpec.java]: Done.

File: clouddriver-ecs/src/integration/java/com/netflix/spinnaker/clouddriver/ecs/test/CreateServerGroupSpec.java [hunk 4]
```diff
@@ -66,6 +66,76 @@ public void createServerGroupOperationTest() throws IOException {
...
+    String requestBody =
+        generateStringFromTestFile(
+            "/createServerGroupOperation-inputs-fargate-NewTargetGroup.json");
```
[the remark this answers is not in the document]
Comment (paragbhingre) [reply clouddriver-ecs/src/integration/java/com/netflix/spinnaker/clouddriver/ecs/test/CreateServerGroupSpec.java]: Changed the file name to "createServerGroupOperation-inputs-fargate-targetGroupMappings.json"

File: clouddriver-ecs/src/integration/resources/testoperations/createServerGroupOperation-inputs-fargate-NewTargetGroup.json (hunk 1 above)
Comment (paragbhingre) [reply clouddriver-ecs/src/integration/resources/testoperations/createServerGroupOperation-inputs-fargate-NewTargetGroup.json]: Source is not required. It was added as a next part of the testing but not really required for this test.

File: clouddriver-ecs/src/integration/java/com/netflix/spinnaker/clouddriver/ecs/test/CreateServerGroupSpec.java [hunk 5]
```diff
@@ -30,25 +30,27 @@
...
+/**
+ * TODO (allisaurus): Ideally these tests would go further and actually assert that the resulting
+ * ecs:create-service call is formed as expected, but for now, it asserts that the given operation
+ * is correctly validated and submitted as a task.
+ */
 public class CreateServerGroupSpec extends EcsSpec {
 
   @Autowired AccountCredentialsRepository accountCredentialsRepository;
 
+  String path = "/ecs/ops/createServerGroup";
```
Comment (allisaurus) [inline clouddriver-ecs/src/integration/java/com/netflix/spinnaker/clouddriver/ecs/test/CreateServerGroupSpec.java:42]: can we name and scope this var [as we do elsewhere](https://github.com/spinnaker/clouddriver/blob/master/clouddriver-ecs/src/integration/java/com/netflix/spinnaker/clouddriver/ecs/EcsSpec.java#L52) in this pkg? I.e.:
`private static final String CREATE_SG_TEST_PATH = ...`

**Review by allisaurus — APPROVED** [review]

**Pull request merged**